src/istore.c

```diff
@@ -976,3 +976,73 @@ istore_svals(PG_FUNCTION_ARGS)
 
     SRF_RETURN_DONE(funcctx);
 }
+
+/*
+ * A boolean function that supports `select '1=>1'::istore = '2=>2'::istore`
+ */
+PG_FUNCTION_INFO_V1(istore_eq);
+Datum
+istore_eq(PG_FUNCTION_ARGS)
+{
+    IStore *is1, *is2;
+    IStorePair *pairs1, *pairs2;
+    int32 len;
+
+    if (PG_ARGISNULL(0) || PG_ARGISNULL(1)) PG_RETURN_NULL();
```

Comment:
Since `NULL` is not a fine result here either, I'd say we rather want to throw a proper error here – just like it is done for aggregation functions when they are called in a non-aggregation context.